### PR TITLE
[MO] Remove IR frontend from available frontend list in MO (#10798)

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** MO should not list IR frontend in a list of available frontends and should not employ it for further IR processing since IR frontend is uncapable to do conversion to the latest IR version.

Master PR https://github.com/openvinotoolkit/openvino/pull/10798

**Ticket:** 80863

### CODE REVIEW

**Comment by rkazants** [comment]
Need approval from the committee before merge

**Review by slyalin — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**
